### PR TITLE
Docs on backup job workflow and architecture

### PR DESCRIPTION
**Pull request by kathancox**
Fixes [DOC-1924](https://cockroachlabs.atlassian.net/browse/DOC-1924)

Added conceptual documentation on how a backup works + diagrams.

### CODE REVIEW

**Comment by kathancox** [comment]
Hey @livlobo @dt @adityamaru , I've included you all for reviews on this page as you all contributed to the sessions on how this works. Would be grateful for all your feedback here! Also, your thoughts on titling the page. Thanks!

**Review by livlobo — COMMENTED** [review]
Hi Kathryn - Overall comment - I found it a little hard to follow the information and found that some terminology  may not be obvious to non-CRL folks(backup detail, descriptors, etc). I have a few suggestions, feel free to use what makes sense to you:

1. Could we setup an example scenario e.g. Let's say you are backing up your cluster with 3 nodes, it has the following ranges R1, R2, R3 on the following nodes N1, N2, N3.. something like that?

2. Maybe we can explain what the "job detail" is by giving examples of the type of info it stores?

3. Maybe A diagram of the overall architecture to start? even if the components don't have a 1:1 mapping with components and interactions in the real architecture they will serve to explain the overall flow. Something like this -  https://weacast.github.io/weacast-docs/architecture/dynamic-view.html

**Comment by kathancox** [comment]
> Hi Kathryn - Overall comment - I found it a little hard to follow the information and found that some terminology may not be obvious to non-CRL folks(backup detail, descriptors, etc). I have a few suggestions, feel free to use what makes sense to you:
> 
> 1. Could we setup an example scenario e.g. Let's say you are backing up your cluster with 3 nodes, it has the following ranges R1, R2, R3 on the following nodes N1, N2, N3.. something like that?
> 2. Maybe we can explain what the "job detail" is by giving examples of the type of info it stores?
> 3. Maybe A diagram of the overall architecture to start? even if the components don't have a 1:1 mapping with components and interactions in the real architecture they will serve to explain the overall flow. Something like this -  https://weacast.github.io/weacast-docs/architecture/dynamic-view.html

Thanks for the review @livlobo, I understand. I'll rework this some and send a revision to you soon!

**Comment by kathancox** [comment]
Thoughts on titling of the page in reviews would be appreciated. My feeling is that Architecture could be a bit overloaded? Could go with something like "How a Backup works" instead.

File: v22.1/backup-architecture.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+The following diagram illustrates the flow from `BACKUP` statement through to a complete backup in cloud storage:
+
+<img src="{{ 'images/v22.1/backup-overview.png' | relative_url }}" alt="A flow diagram representing the process of a backup job from statement through to backup data stored." style="border:0px solid #eee;max-width:100%" />
+
+### Glossary 
+
+Term         | Description
+-------------+-----------------------------
+Coordinator  | The node that claims the job and coordinates the distributed work for the backup job.
```
[the remark this answers is not in the document]
Comment (dt) [reply v22.1/backup-architecture.md:31]: If we did that, I think the structure of this changes to be "Step one is to make a backup job from the backup statement. this requires validating the options and checking permissions, then creating the job record that reflects what to backup, where to back it up, and any other options. After this, if the user requested a "detached" backup the statement is complete, as it has created an uncommitted but otherwise ready-to-run backup job. If they did not request detached, this job is committed, then the statement waits for the job to be started, run and terminate to then return the result of the job execution as the statement's result."

File: v22.1/backup-architecture.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ ~~~
+        job_id
+----------------------
+  592786066399264769
+(1 row)
+ ~~~
+
+As soon as the detail of the backup is written to the job record, the planning phase’s transaction will commit and the nodes can claim the backup job.
+
+Clusters have system tables that are visible to each of their nodes. One of these tables, the `system.jobs` table, is a list of all adoptable jobs. Each row in this table represents a job, which is described by the job record created in the backup's planning phase.
```
Comment (dt) [inline v22.1/backup-architecture.md:63]: mentioned above, but these next couple lines seem like they _might_ be better moved to a general "how jobs work" doc?

File: v22.1/backup-architecture.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+Once the statement has been passed through the planning phase a `job ID` output will return, which represents that the job has started executing.
+
+ ~~~
+        job_id
+----------------------
+  592786066399264769
+(1 row)
+ ~~~
+
+As soon as the detail of the backup is written to the job record, the planning phase’s transaction will commit and the nodes can claim the backup job.
```
[the remark this answers is not in the document]
Comment (dt) [reply v22.1/backup-architecture.md:61]: What _is_ maybe relevant to mention is that once the job record is committed, the cluster will try to run that backup even if the client disconnects or the node they sent the backup statement to terminates; it is now a persisted job that any node in the cluster can take over executing it to ensure it runs even if individual nodes don't.

File: v22.1/backup-architecture.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+- Map out the directory to which the nodes will write.
+- Gather all the leaseholders of the keys to be backed up.
+
+To map out the directory to which the nodes will write, the coordinator identifies the [type](backup-and-restore-overview.html#backup-and-restore-types) of backup. This determines the name of the new (or edited) directory to store the backup files in. For example, if there is a full backup existing in the target storage location, the upcoming backup will be [incremental](take-full-and-incremental-backups.html#incremental-backups) and therefore append to the [full](take-full-and-incremental-backups.html#full-backups) backup.
+
+From the cluster, the coordinator will gather all the [leaseholders](architecture/reads-and-writes-overview.html) of the keys that will be part of the backup. The leaseholder is responsible for serving the data from the storage layer because all [writes](architecture/reads-and-writes-overview.html#write-scenario) go through a leaseholder. Since any node in a cluster can become the coordinator and all nodes could be responsible for exporting data during a backup, it is necessary that all nodes can connect to the storage location.
```
[the remark this answers is not in the document]
Comment (dt) [reply v22.1/backup-architecture.md:81]: More concretely, I think I'd break this section into sub-sections for span and time range resolution phase, distributed data exporting phase, then metadata writing conclusion phase.

File: v22.1/backup-architecture.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+While processing, the nodes write checkpoint progress to track their backup work. The coordinating node will aggregate the progress data into checkpoint files in the storage bucket. The checkpoint files provide a marker for the backup to resume after a retryable state, such as when it has been paused.
+
+The coordinator will begin to write the backup manifest file only after each of the nodes have completed their work. In the diagram, **Node 2** is exporting the backup data for **R1** as its leaseholder as well as exporting the backup's metadata as the coordinator.
+
+The backup manifest file is a metadata representation of everything a backup contains. That is, all the information a [restore](restore.html) job will need to complete successfully. A backup without a manifest file would indicate that the backup did not complete properly and would not be restorable.
+
+With multiple backups complete, the specified storage location will contain a set of backups, which CockroachDB defines as a [_backup collection_](take-full-and-incremental-backups.html#backup-collections).
```
Comment (dt) [inline v22.1/backup-architecture.md:93]: This line could be a little confusing: are we talking about _a_ backup to _a_ location, or are we talking about backing up to a _collection_ repeatedly, i.e. performing subsequent backups? architecturally, those subsequent backups are to different locations i.e. unique paths within the collection. 

I think I'd maybe just make a whole separate section for collections?

**Comment by kathancox** [comment]
Hey @dt I've revised the page somewhat incorporating your feedback. One of the things I wasn't 100% on was whether I've placed the key and time range resolution in the right place — within the main "Resolution" heading. 

I'd appreciate another review on the overall changes here to make sure it's heading in the right direction!

File: v22.1/backup-architecture.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+To map out the storage location's directory to which the nodes will write the data, the coordinator identifies the [type](backup-and-restore-overview.html#backup-and-restore-types) of backup. This determines the name of the new (or edited) directory to store the backup files in. For example, if there is a full backup existing in the target storage location, the upcoming backup will be [incremental](take-full-and-incremental-backups.html#incremental-backups) and therefore append to the full backup. For more information on how CockroachDB structures backups in storage, see [Backup collections](take-full-and-incremental-backups.html#backup-collections).
+
+### Key and time range resolution
+
+In this part of the resolution phase, the coordinator will calculate all the necessary spans of keys and their time range that the cluster needs to export for this backup. It divides the key spans based on which node is the range [leaseholder](architecture/overview.html#architecture-leaseholder) for that key. For this backup, each leaseholder node is responsible for:
+
+1. Asking the [storage layer](architecture/storage-layer.html) for the content of each key.
```
Comment (dt) [inline v22.1/backup-architecture.md:61]: each key span?

File: v22.1/backup-architecture.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+- Provide a plan to the leaseholder nodes.
+
+To map out the storage location's directory to which the nodes will write the data, the coordinator identifies the [type](backup-and-restore-overview.html#backup-and-restore-types) of backup. This determines the name of the new (or edited) directory to store the backup files in. For example, if there is a full backup existing in the target storage location, the upcoming backup will be [incremental](take-full-and-incremental-backups.html#incremental-backups) and therefore append to the full backup. For more information on how CockroachDB structures backups in storage, see [Backup collections](take-full-and-incremental-backups.html#backup-collections).
+
+### Key and time range resolution
+
+In this part of the resolution phase, the coordinator will calculate all the necessary spans of keys and their time range that the cluster needs to export for this backup. It divides the key spans based on which node is the range [leaseholder](architecture/overview.html#architecture-leaseholder) for that key. For this backup, each leaseholder node is responsible for:
```
Comment (dt) [inline v22.1/backup-architecture.md:59]: based on which node is *the leaseholder of* the range for that key *span* ?

also it isn't technically the leaseholder who is responsible, for the following steps but rather a backup sql processor which is running on the node which is the leaseholder, or maybe on a different node e.g. in serverless or if things moved after planning.

File: v22.1/backup-architecture.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+1. Asking the [storage layer](architecture/storage-layer.html) for the content of each key.
+1. Receiving the content from the storage layer. 
+1. Writing it to the backup storage location or [locality-specific location](take-and-restore-locality-aware-backups.html) (whichever locality best matches the node). 
+
+Since any node in a cluster can become the coordinator and all nodes could be responsible for exporting data during a backup, it is necessary that all nodes can connect to the backup storage location.
+
+## Export phase 
+
+Once the coordinator has provided a plan to each of the leaseholders that determines the backup data to process, the distributed export of the backup data begins.
```
Comment (dt) [inline v22.1/backup-architecture.md:69]: pedantic again, but "each of the backup processors, ..."

File: v22.1/backup-architecture.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+The backup export phase has two distinct parts:
```
Comment (dt) [inline v22.1/backup-architecture.md:71]: Arguably metadata writing is a wholly separate phase? like, we have to be fully done exporting across all processors before the coordinator can write the manifest/stats/etc.

File: v22.1/backup-architecture.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+- [Distributed data exporting](#distributed-data-exporting)
+- [Metadata writing](#metadata-writing)
+
+### Distributed data exporting
+
+In the following diagram, **Node 2** and **Node 3** contain the leaseholders for the **R1** and **R2** [ranges](architecture/overview.html#architecture-range). Therefore, for the example backup job, these nodes will export the backup data to the specified storage location. While processing, the nodes write checkpoint progress to track their backup work. The coordinator node will aggregate the progress data into checkpoint files in the storage bucket. The checkpoint files provide a marker for the backup to resume after a retryable state, such as when it has been paused.
```
Comment (dt) [inline v22.1/backup-architecture.md:78]: "for the example backups job, the _backup processors on_ these nodes..."

also, the individual nodes do not write checkpoints; they return progress metadata to the coordinator who merges it to write the checkpoints as the next sentence says, so perhaps here it should just say "emit" or "return" progress information instead of "Write checkpoint" .

File: v22.1/backup-architecture.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+<img src="{{ 'images/v22.1/backup-processing.png' | relative_url }}" alt="Three-node cluster exporting backup data from the leaseholders" style="border:0px solid #eee;max-width:100%" />
+
+### Metadata writing
+
+To conclude the backup job, the coordinator will begin to write the backup metadata files only after each of the nodes have completed their data export work. In the diagram, **Node 2** is exporting the backup data for **R1** as that range's leaseholder, but this node also exports the backup's metadata as the coordinator.
+
+The backup metadata files represent everything a backup contains. That is, all the information a [restore](restore.html) job will need to complete successfully. A backup without metadata files would indicate that the backup did not complete properly and would not be restorable.
```
Comment (dt) [inline v22.1/backup-architecture.md:86]: s/represent/describe/ ?

**Review by dt — APPROVED** [review]
Wow, this looks really good! like, really good!

I think we _could_ move the bulleted list that starts with "Each of the node's backup SQL processors are responsible for:" to just below the `## Export phase ` heading since it is talking about what happens during export, but we could also leave it as is, since it is talking about what _will_ happen during export which is why we do the planning the way we do. I'm good with either.

Again though, overall looks fantastic.

File: v22.1/backup-architecture.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+## Resolution phase 
+
+Once one of the nodes has claimed the job from the system job table, it will take the job record’s information and outline a plan. This node becomes the _coordinator_. In our example, **Node 2** becomes the coordinator and starts to complete the following to prepare and resolve the targets for distributed backup work: 
+
+- Test the connection to the storage bucket URL (`'s3://bucket'`).
+- Determine the specific subdirectory for this backup, including if it should be incremental from any discovered existing directories.
+- Calculate the keys and time range (if incremental) of the backup data.
+- Determine the [leaseholder](architecture/overview.html#architecture-leaseholder) nodes for the keys to back up. 
+- Provide a plan to the notes that will execute the data export (typically the leaseholder node).
```
Comment (livlobo) [inline v22.1/backup-architecture.md:53]: typo: Provide a plan to the nodes..

File: v22.1/backup-architecture.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+1. Receiving the content from the storage layer. 
+1. Writing it to the backup storage location or [locality-specific location](take-and-restore-locality-aware-backups.html) (whichever locality best matches the node). 
+
+Since any node in a cluster can become the coordinator and all nodes could be responsible for exporting data during a backup, it is necessary that all nodes can connect to the backup storage location.
+
+## Export phase 
+
+Once the coordinator has provided a plan to each of the backup SQL processors that specifies the backup data, the distributed export of the backup data begins.
+
+In the following diagram, **Node 2** and **Node 3** contain the leaseholders for the **R1** and **R2** [ranges](architecture/overview.html#architecture-range). Therefore, in this example backup job, the backup data will be exported from these nodes to the specified storage location. While processing, the nodes emit progress data to track their backup work. The coordinator node will aggregate the progress data into checkpoint files in the storage bucket. The checkpoint files provide a marker for the backup to resume after a retryable state, such as when it has been paused.
```
Comment (livlobo) [inline v22.1/backup-architecture.md:75]: > While processing, the nodes emit progress data to track their backup work. The coordinator node will aggregate the progress data into checkpoint files in the storage bucket. 
 
 How is this emitted? Do N1 and N3 send this to N2, and N2 aggregates it?

**Review by livlobo — APPROVED** [review]
This looks great, @kathancox ! It has the right level of abstraction. I've added some minor nits, but otherwise good to merge!

File: v22.1/backup-architecture.md (hunk 13 above)
Comment (kathancox) [reply v22.1/backup-architecture.md]: Good point, that was slightly unclear. I think I have cleared this up and referred to the diagram here for the information.

**Review by andf-crl — APPROVED** [review]
To echo earlier reviewers' sentiments also, this is downright fantastic: a clear, straightforward read on a complicated, multi-stage, intricate process! I've left small optional language suggestions for clarity, but otherwise LGTM! Wow!

If you are feeling adventurous, you could also link to this excellent new resource from somewhere on the `Backup and Restore Overview` page!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14238#-:-N8_JfPbDZxr6LuFbwez:b-xg0i23)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @dt, @kathancox, and @livlobo)

___
*[`v22.1/backup-architecture.md` line 91 at r1](https://reviewable.io/reviews/cockroachdb/docs/14238#-N47f5skX_dmMUWAZT29-r1-91:-N8_IiD29D2F4J7H7VaH:br9pi5g) ([raw file](https://github.com/cockroachdb/docs/blob/fadc01e1e749cca2f9983f420ee5ac53d7cd589f/v22.1/backup-architecture.md#L91)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

there are multiple files that make up the manifest these days: there is _the_ manifest, plus an SST version of it, plus various stats files. I wonder if we'd be better just collectively referring to "metadata files" to avoid being too specific that it becomes stale?
</blockquote></details>

This sentence is supremely helpful, and answers several user questions before they are asked. Nice work!
___
*[`v22.1/backup-architecture.md` line 41 at r2](https://reviewable.io/reviews/cockroachdb/docs/14238#-N47f5skX_dmMUWAZT29-r2-41:-N8_EQDK6QxRZIQsAaYg:b-91umzk) ([raw file](https://github.com/cockroachdb/docs/blob/174c80a09086f9da518167cbc3a926e02c281b71/v22.1/backup-architecture.md#L41)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

maybe "run _as described in below_, and terminate..." ?
</blockquote></details>

optional nit: `statement waits for the backup job to start ... to the then return the results` --> `statement waits to return the results until the backup job starts, runs (as described), and terminates.`
___
*[`v22.1/backup-architecture.md` line 8 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_Dn4lAGbJH5yxxoED:-N8_Dn4lAGbJH5yxxoEE:b-6r2h84) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L8)):*
> ```Markdown
> ---
> 
> CockroachDB backups operate as _jobs_, which are potentially long-running operations that could span multiple SQL sessions. Unlike regular SQL statements, which CockroachDB routes to the [optimizer](cost-based-optimizer.html) for processing, a [`BACKUP`](backup.html) statement will move into a job workflow. A backup job has four main phases: [job creation](#job-creation-phase), [resolution](#resolution-phase), [export data](#export-phase), and [metadata writing](#metadata-writing-phase).
> ```

- supernit: I would make the 4 steps numbered list elements, since they become dedicated headers each, and this would somewhat match the right-TOC also. The four elements seem important enough to merit some attention-grabbing whitespace that accompanies list elements too!
___
*[`v22.1/backup-architecture.md` line 14 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_BgfiCU_wRg7M7CHy:-N8_BgfiCU_wRg7M7CHz:bdj2wpn) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L14)):*
> ```Markdown
> ## Overview
> 
> A high-level workflow of a backup:
> ```

Clarification: Consider explicitly specifying the actor here: CockroachDB does all these things, not the user. "Validate the parameters" almost feels like a directive in this context. Maybe something like:

At a high level, CockroachDB performs the following task when running a backup job:
1. Validates ... then writes
1. ... determines ...
1. Instructs ...
1. Records ...

Or it could be spliced into the  numbered entires themselves if you prefer. Just think it would be helpful to clarify that CockroachDB does all of this for you, user.
___
*[`v22.1/backup-architecture.md` line 23 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_DZpg26s0JebKrdf_:-N8_DZpg26s0JebKrdfa:b-9ha3tr) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L23)):*
> ```Markdown
> The following diagram illustrates the flow from `BACKUP` statement through to a complete backup in cloud storage:
> 
> A flow diagram representing the process of a backup job from statement through to backup data stored.
> ```

This layout is amazing, super-easy to follow top-> bottom, left->right for an otherwise-complicated and staged process. Nice work here!!
___
*[`v22.1/backup-architecture.md` line 43 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_EzRl2Si1sQZ_sxBF:-N8_EzRm0oXLrekmmnJ5:b-o55ijb) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L43)):*
> ```Markdown
> If a [`detached`](backup.html#detached) backup was requested, the `BACKUP` statement is complete as it has created an uncommitted, but otherwise ready-to-run backup job. You'll find the job ID returned as output. Without the `detached` option, the job is committed and the statement waits for the backup job to start, run (as described in the following sections), and terminate to then return the results.
>  
> Once the job record is committed, the cluster will try to run the backup job even if a client disconnects or the node handling the `BACKUP` statement terminates. From this point, the backup is a persisted job that any node in the cluster can take over executing to ensure it runs. The job record will move to the system jobs table ready for adoption. 
> ```

nit: `move to the system jobs table ready for adoption.` --> `move to the system jobs table, ready for a node to claim it.`

or maybe just:

`move to the system jobs table, ready for adoption by an eligible node.`
___
*[`v22.1/backup-architecture.md` line 47 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_FY5u2lLvCzwCtt1Z:-N8_FY5u2lLvCzwCtt1_:b-cjqp9y) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L47)):*
> ```Markdown
> ## Resolution phase 
> 
> Once one of the nodes has claimed the job from the system job table, it will take the job record’s information and outline a plan. This node becomes the _coordinator_. In our example, **Node 2** becomes the coordinator and starts to complete the following to prepare and resolve the targets for distributed backup work: 
> ```

nit: `system jobs table` (plural, to match elsewhere)
___
*[`v22.1/backup-architecture.md` line 51 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_Fp5v39pFb7CLwT66:-N8_Fp5v39pFb7CLwT67:b3qxn68) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L51)):*
> ```Markdown
> - Test the connection to the storage bucket URL (`'s3://bucket'`).
> - Determine the specific subdirectory for this backup, including if it should be incremental from any discovered existing directories.
> - Calculate the keys and time range (if incremental) of the backup data.
> ```

nit: Could be construed that both keys and time ranges are affected by whether the backup is incremental or not. If not, consider something like:

`Calculate the keys of the backup data, as well as the time range if the backup is incremental.`
___
*[`v22.1/backup-architecture.md` line 55 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_Gc282oMdrTk4FM2g:-N8_Gc282oMdrTk4FM2h:b-xn8p0b) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L55)):*
> ```Markdown
> - Provide a plan to the nodes that will execute the data export (typically the leaseholder node).
> 
> To map out the storage location's directory to which the nodes will write the data, the coordinator identifies the [type](backup-and-restore-overview.html#backup-and-restore-types) of backup. This determines the name of the new (or edited) directory to store the backup files in. For example, if there is an existing full backup in the target storage location, the upcoming backup will be [incremental](take-full-and-incremental-backups.html#incremental-backups) and therefore append to the full backup after any existing incremental layers discovered in it. 
> ```

nit: `time range` -- > `time ranges`
___
*[`v22.1/backup-architecture.md` line 77 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_Ht9m6-X-4PAuTNqV:-N8_Ht9m6-X-4PAuTNqW:bffhunk) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L77)):*
> ```Markdown
> In the following diagram, **Node 2** and **Node 3** contain the leaseholders for the **R1** and **R2** [ranges](architecture/overview.html#architecture-range). Therefore, in this example backup job, the backup data will be exported from these nodes to the specified storage location. 
> 
> While processing, the nodes emit progress data that tracks their backup work to the coordinator. In the diagram, **Node 3** will send progress data to **Node 2**. The coordinator node will then aggregate the progress data into checkpoint files in the storage bucket. The checkpoint files provide a marker for the backup to resume after a retryable state, such as when it has been paused.
> ```

We could perhaps link `paused` here: https://www.cockroachlabs.com/docs/stable/pause-job.html esp because we use the jobs framework to manage backups instead of something backup-specific, which the user might otherwise be able to find on their own.
___
*[`v22.1/backup-architecture.md` line 83 at r4](https://reviewable.io/reviews/cockroachdb/docs/14238#-N8_ILU2CD69Xai-84BP:-N8_ILU2CD69Xai-84BQ:b-12f1sv) ([raw file](https://github.com/cockroachdb/docs/blob/e5943fd2e3b860142221684d68cf8a2dab2d9efa/v22.1/backup-architecture.md#L83)):*
> ```Markdown
> ## Metadata writing phase
> 
> To conclude the backup job, the coordinator will begin to write the backup metadata files only after each of the nodes have completed their data export work. In the diagram, **Node 2** is exporting the backup data for **R1** as that range's leaseholder, but this node also exports the backup's metadata as the coordinator.
> ```

nit: consider inverting the order here: `Once each of the nodes have fully completed their data export work, the coordinator will conclude the backup job by writing the backup metadata files.`

**Comment by kathancox** [comment]
@andf-crl Thanks so much! Your docs reviews are super, super helpful 😺 I have incorporated all your suggestions.

**Comment by kathancox** [comment]
Hey @dt & @livlobo Thanks a lot for the super detailed reviews and support on this one, really appreciated. Also, thanks @adityamaru for walking me through the backup process some months ago!
